### PR TITLE
fix(invitedUsers): filter deleted users for invitedUsers endpoint

### PR DESCRIPTION
**Pull request by Phil91**
## Description

Remove the deleted users from the invitedUsers request

## Why

Deleted users don't have an userEntityId, therefor the request will fail when keycloak is requested for the user informations

## Issue

N/A - Jira Issue: CPLP-2986

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Repositories/InvitationRepository.cs [hunk 1]
```diff
@@ -35,15 +36,20 @@ public InvitationRepository(PortalDbContext dbContext)
...
-        (from invitation in _dbContext.Invitations
-         join invitationStatus in _dbContext.InvitationStatuses on invitation.InvitationStatusId equals invitationStatus.Id
-         join companyuser in _dbContext.CompanyUsers on invitation.CompanyUserId equals companyuser.Id
-         where invitation.CompanyApplicationId == applicationId
-         select new InvitedUserDetail(
-             companyuser.Identity!.UserEntityId,
-             invitationStatus.Id,
-             companyuser.Email
-         ))
+        _dbContext.Invitations
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/InvitationRepository.cs:39]: I suggest:
```
    public IAsyncEnumerable<InvitedUserDetail> GetInvitedUserDetailsUntrackedAsync(Guid applicationId) =>
        _dbContext.Invitations
            .AsNoTracking()
            .Where(invitation => invitation.CompanyApplicationId == applicationId && invitation.CompanyUser!.Identity!.UserStatusId != UserStatusId.DELETED)
            .Select(invitation => new InvitedUserDetail(
                invitation.CompanyUser!.Identity!.UserEntityId,
                invitation.InvitationStatusId,
                invitation.CompanyUser.Email))
            .AsAsyncEnumerable();
```

File: tests/portalbackend/PortalBackend.DBAccess.Tests/InvitationRepositoryTests.cs [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+    }
+
+    [Fact]
+    public async Task GetInvitedUserDetailsUntrackedAsync_WithValid_ReturnsExpected()
+    {
+        var sut = await CreateSut().ConfigureAwait(false);
+
+        var result = await sut.GetInvitedUserDetailsUntrackedAsync(_applicationId).ToListAsync().ConfigureAwait(false);
+
+        result.Should().NotBeNull();
```
Comment (ntruchsess) [inline tests/portalbackend/PortalBackend.DBAccess.Tests/InvitationRepositoryTests.cs:30]: I suggest:
```
        result.Should().HaveCount(2)
            .And.Satisfy(
                x => x.EmailId == "test@user.com" && x.InvitationStatus == InvitationStatusId.CREATED,
                x => x.EmailId == "company.admin1@acme.corp" && x.InvitationStatus == InvitationStatusId.CREATED);
```
(there's no point in asserting 'NotBeNull()' if there are more asserts that implicitly fail if NotBeNull is not satisfied)

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: tests/portalbackend/PortalBackend.DBAccess.Tests/InvitationRepositoryTests.cs (hunk 2 above)
Comment (Phil91) [reply tests/portalbackend/PortalBackend.DBAccess.Tests/InvitationRepositoryTests.cs]: adjusted

File: src/portalbackend/PortalBackend.DBAccess/Repositories/InvitationRepository.cs (hunk 1 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/InvitationRepository.cs:38]: adjusted

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**